SpatialGDK/Source/SpatialGDK/Private/SpatialView/OpList/EntityComponentOpList.cpp

```diff
@@ -78,24 +80,117 @@ EntityComponentOpListBuilder& EntityComponentOpListBuilder::SetAuthority(Worker_
+	Op.op.reserve_entity_ids_response.status_code = StatusCode;
+	Op.op.reserve_entity_ids_response.message = StoreString(MoveTemp(Message));
+	OpListData->Ops.Add(Op);
+	return *this;
+}
+
+EntityComponentOpListBuilder& EntityComponentOpListBuilder::AddDeleteEntityCommandResponse(Worker_EntityId EntityID,
+																						   Worker_RequestId RequestId,
+																						   Worker_StatusCode StatusCode,
+																						   StringStorage Message)
+{
+	Worker_Op Op = {};
+	Op.op_type = WORKER_OP_TYPE_DELETE_ENTITY_RESPONSE;
+	Op.op.delete_entity_response.entity_id = EntityID;
+	Op.op.delete_entity_response.request_id = RequestId;
+	Op.op.delete_entity_response.status_code = StatusCode;
+	Op.op.delete_entity_response.message = StoreString(MoveTemp(Message));
+	OpListData->Ops.Add(Op);
+	return *this;
+}
+
+EntityComponentOpListBuilder& EntityComponentOpListBuilder::AddEntityQueryCommandResponse(Worker_RequestId RequestId,
+																						  TArray<OpListEntity> Results,
+																						  Worker_StatusCode StatusCode,
+																						  StringStorage Message)
+{
+	Worker_Op Op = {};
+	Op.op_type = WORKER_OP_TYPE_ENTITY_QUERY_RESPONSE;
+	Op.op.entity_query_response.result_count = Results.Num();
+	Op.op.entity_query_response.results = StoreQueriedEntities(MoveTemp(Results));
+	Op.op.entity_query_response.request_id = RequestId;
+	Op.op.entity_query_response.status_code = StatusCode;
+	Op.op.entity_query_response.message = StoreString(MoveTemp(Message));
 	OpListData->Ops.Add(Op);
 	return *this;
 }
 
+EntityComponentOpListBuilder& EntityComponentOpListBuilder::AddEntityCommandResponse(Worker_EntityId EntityID, Worker_RequestId RequestId,
+																					 Worker_StatusCode StatusCode, StringStorage Message)
+{
+	Worker_Op Op = {};
+	Op.op_type = WORKER_OP_TYPE_COMMAND_RESPONSE;
+	Op.op.command_response.entity_id = EntityID;
+	Op.op.command_response.request_id = RequestId;
+	Op.op.command_response.status_code = StatusCode;
+	Op.op.command_response.message = StoreString(MoveTemp(Message));
+	OpListData->Ops.Add(Op);
+	return *this;
+}
+
+const char* EntityComponentOpListBuilder::StoreString(StringStorage Message) const
+{
+	OpListData->MessageStorage.Add(MoveTemp(Message));
+	return OpListData->MessageStorage.Last().Get();
+}
+
+const Worker_Entity* EntityComponentOpListBuilder::StoreQueriedEntities(TArray<OpListEntity> Entities) const
+{
+	OpListData->QueriedEntities.Push(TArray<Worker_Entity>());
+	for (auto& Entity : Entities)
+	{
+		Worker_Entity CurrentEntity;
+		CurrentEntity.entity_id = Entity.EntityId;
+		OpListData->QueriedComponents.Push(TArray<Worker_ComponentData>());
+		for (auto& Component : Entity.Components)
+		{
+			OpListData->DataStorage.Add(MoveTemp(Component));
+			OpListData->QueriedComponents.Last().Push(OpListData->DataStorage.Last().GetWorkerComponentData());
+		}
+
+		CurrentEntity.components = OpListData->QueriedComponents.Last().GetData();
```

Comment:
Would grab a reference to last when you add the array in the first place. Then use that ref instead of calling last. Makes the calls to last fractionally easier to read